src/bam/mod.rs

```diff
@@ -301,6 +316,30 @@ impl IndexedReader {
             })
         }
     }
+    /// Create a new Reader.
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - the path. Use "-" for stdin.
+    /// * `index_path` - the index path to use
+    fn new_with_index_path(
```

Comment:
Just a minor style thing: want to add something like `<P: &ffi::CStr>` to remove type redundancy in the argument definitions? Or am I missing something and this doesn't work here?

Reply:
I'm not firm in Rust yet , but 
```fn new_with_index_path<P:&ffi::CStr> (```
get's the compile mad at me because it doesn't like the & in that position.

Reply:
I'm not too firm either., and just realised that the construct I suggested is only for specifying generic functions with some Trait requirements. I.e. if I have now grasped this correctly, in the <P: AsRef<Path>> of the other functions, this means that this function will work for any Type P that implements the struct Path and for Path has an implementation of the Trait `AsRef`. So sorry for the noise, just ignore my suggestion.

Reply:
Ah, normally I think you'd have to do `fn foo<P: X>(thing: &X) -> {...}`. However, X has to be a trait and in this case `ffi::CStr` is a struct.